src/components/SocialMediaLinks/SocialMediaLinks.test.tsx

```diff
@@ -0,0 +1,476 @@
+import { render, screen } from "@testing-library/react";
+import { axe } from "jest-axe";
+import * as React from "react";
+import renderer from "react-test-renderer";
+import SocialMediaLinks from "./SocialMediaLinks";
+import { socialMediaDataMap } from "./SocialMediaDataMap";
+
+// If you want to see what's happening.
+// screen.debug();
+
+describe("SocialMediaLinks Accessibility", () => {
+  describe("Labels", () => {
+    it("passes axe accessibility test without visible labels", async () => {
+      const { container } = render(<SocialMediaLinks />);
+      expect(await axe(container)).toHaveNoViolations();
+    });
+    it("passes axe accessibility test with labels set to true", async () => {
+      const { container } = render(<SocialMediaLinks showLabels={true} />);
+      expect(await axe(container)).toHaveNoViolations();
+    });
+  });
+  describe("Colors", () => {
+    it("passes axe accessibility test with link color & labels", async () => {
+      const { container } = render(
+        <SocialMediaLinks showLabels={true} color={"link"} />
+      );
+      expect(await axe(container)).toHaveNoViolations();
+    });
+    it("passes axe accessibility test with textInverse color & labels", async () => {
+      const { container } = render(
+        <SocialMediaLinks showLabels={true} color={"textInverse"} />
+      );
+      expect(await axe(container)).toHaveNoViolations();
+    });
+  });
+  describe("Borders", () => {
+    it("passes axe accessibility test with straight borders & labels", async () => {
+      const { container } = render(
+        <SocialMediaLinks showLabels={true} border={"straight"} />
+      );
+      expect(await axe(container)).toHaveNoViolations();
+    });
+    it("passes axe accessibility test with circular borders", async () => {
+      const { container } = render(<SocialMediaLinks border={"circular"} />);
+      expect(await axe(container)).toHaveNoViolations();
+    });
+  });
+});
+
+describe("SocialMediaLinks", () => {
+  it("Renders default", () => {
+    render(<SocialMediaLinks />);
+
+    const links = screen.getAllByRole("link");
+
+    // All the links should be there.
+    // @todo I dunno. Isn't this kinda cheating? It will always be true, so why bother?
+    //   But I am trying to avoid changing the test if/when the socialMediaDataMap gets updated.
+    expect(links).toHaveLength(socialMediaDataMap.length);
+
+    // The links should be what we assigned, above.
+    // @todo Again, I could do this most easily by looping through the links array and mapping
+    //   each type to the socialMediaDataMap to ensure there is a match. But what is the point?
+    //   It will never fail since the data sets are both from the same source.
+    //   Are we doing this just to demonstrate that the component is rendered at all?
+    //   Don't the other accessibility tests and snapshots do that?
```

Comment:
The accessibility is an overview of the entire component but it can't read or verify the URLs. It just makes sure that the DOM is correct but not that the href value is the correct one for the specific social media type. The snapshots tests the visual aspects and DOM structure which is close to unit testing but can vary. For example, if the label and icon were switched in order, the unit test would not catch that because it cares about those values being in the document but not where or the order. The snapshot would fail and say, "hey, I expect the label to go before the icon and now that's not true."

Reply:
OK, but, is there a scenario where the unit test would fail but the snapshot would succeed?

Reply:
Hmm don't think so. The inverse may be true but not what you asked, or for now I can't think of any specific scenario.

Reply:
Since that is the case I would argue that the unit tests are duplicative. But you note above that we should keep them for consistency. So I will keep them. Note that they may require maintenance for any future updates to the component, which is, I gather, unusual.